WordPress/src/main/java/org/wordpress/android/ui/sitecreation/sitename/SiteCreationSiteNameFragment.kt

```diff
@@ -0,0 +1,97 @@
+import org.wordpress.android.util.DisplayUtilsWrapper
+import javax.inject.Inject
+
+/**
+ * Implements the Site Name UI
+ */
+@Suppress("TooManyFunctions")
+class SiteCreationSiteNameFragment : Fragment() {
+    @Inject internal lateinit var viewModelFactory: ViewModelProvider.Factory
+    @Inject internal lateinit var uiHelper: UiHelpers
+    @Inject internal lateinit var displayUtils: DisplayUtilsWrapper
+
+    private lateinit var viewModel: SiteCreationSiteNameViewModel
+    private var binding: SiteCreationSiteNameFragmentBinding? = null
+
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        (requireActivity().applicationContext as WordPress).component().inject(this)
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.site_creation_site_name_fragment, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        viewModel = ViewModelProvider(requireActivity(), viewModelFactory)
+                .get(SiteCreationSiteNameViewModel::class.java)
+
+        val binding = SiteCreationSiteNameFragmentBinding.bind(view)
+        this.binding = binding
+        with(binding) {
+            setupUi()
+            setupViewModel()
+            setupActionListeners()
+        }
+    }
+
+    private fun SiteCreationSiteNameFragmentBinding.setupUi() {
+        siteCreationSiteNameTitlebar.appBarTitle.isInvisible = !isPhoneLandscape()
+    }
+
+    private fun SiteCreationSiteNameFragmentBinding.setupViewModel() {
+        viewModel.onSiteNameEntered.observe(viewLifecycleOwner, (requireActivity() as
+                SiteNameScreenListener)::onSiteNameEntered)
+        viewModel.start()
+    }
+
+    private fun SiteCreationSiteNameFragmentBinding.setupActionListeners() {
+        siteCreationSiteNameTitlebar.skipButton.setOnClickListener { viewModel.onSkipPressed() }
+        siteCreationSiteNameTitlebar.backButton.setOnClickListener { viewModel.onBackPressed() }
+        // TODO: disable continue button when input is empty
+        continueButton.setOnClickListener {
+            viewModel.onSiteNameEntered(input.text.toString())
+        }
+        input.setOnEditorActionListener { _, actionId, _ ->
+            when (actionId) {
+                EditorInfo.IME_ACTION_DONE -> {
+                    viewModel.onSiteNameEntered(input.text.toString())
+                    return@setOnEditorActionListener true
+                }
+                else -> return@setOnEditorActionListener false
+            }
+        }
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        binding = null
+    }
+
+    private fun isPhoneLandscape() = displayUtils.isLandscapeBySize() && !displayUtils.isTablet()
```

Comment:
Seeing that `isPhoneLandscape()` is quite a "popular" need I was thinking maybe we can extract it to `displayUtils`.

Wdyt?

Reply:
Good point Ovi. Not sure whether or not we'll end up needing this, but if we keep it, I will extract it to `displayUtils` :+1: